ops/ansible/playbooks-ccs/roles/platinum/files/permit-user-access

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+set -e
+
+###########################################################
+# This script serves to grant SSH and sudo access to users
+# whose properties and permissions are stored in SSM.
+###########################################################
+
+ENV=$1
+
+# Return users from SSM, filtering results to include only the relevant user data
+# based on their respective EUA codes. AWS Cli doesn't support this type of filtering
+# natively, so we do so via jq.
+get_ssm_users () {
+  user_ssm="/bfd/mgmt/common/sensitive/user/"
+  # SSM path ends with a valid, alphanumeric EUA code
+  eua_identifier_pattern="[A-Z0-9]{4}$"
+  users=$(aws ssm get-parameters-by-path \
+    --with-decryption \
+    --path $user_ssm \
+    --recursive \
+    --region us-east-1 \
+    --query 'Parameters' \
+    | jq -r "map(select(.Name | test(\"^${user_ssm}${eua_identifier_pattern}\")))")
+  echo "$users"
+}
+
+# Grants SSH access to the specified user, if applicable, based
+# on their environment permissions stored in SSM.
+permit_user_ssh () {
+  user=$1
+
+  # Skip the user if they do not have ssh access to the current environment
+  has_env_access=$(echo "$user" | jq -r ".env_access | any(. == \"${ENV}\")")
+  if ! $has_env_access ; then
+    return
+  fi
+
+  user_name=$(echo "$user" | jq -r '.user_name')
+  ssh_public_key=$(echo "$user" | jq -r '.ssh_public_key')
+  user_ssh_dir="/home/$user_name/.ssh"
+
+  adduser --comment '' "$user_name"
+  mkdir -p "$user_ssh_dir"
+  touch "$user_ssh_dir"/authorized_keys
+  echo "$ssh_public_key" > "$user_ssh_dir"/authorized_keys
```

Comment:
The `touch` isn't hurting anything but what value does it add?  Wouldn't the redirection on the next line create it anyway?

Reply:
Fair. Removed :)